### ISSUE TITLE
Show Error Page on Error

### ISSUE THREAD

**Issue by bbinto** [mozilla-mobile/fenix#921]
As a User I want to understand why a website could not be loaded

### Why/User Benefit/User Problem
- Make sure that users understand why a website could not be loaded
- Present them with an error screen and a way to get out / continue browsing to another website

### What / Requirements

This is all about displaying an error page when a page did not load because of an error condition.

We have an [Error Pages](https://github.com/mozilla-mobile/android-components/tree/master/components/browser/errorpages) component that covers the following:

* ERROR_SECURITY_SSL
* ERROR_SECURITY_BAD_CERT
* ERROR_NET_INTERRUPT
* ERROR_NET_TIMEOUT
* ERROR_CONNECTION_REFUSED
* ERROR_UNKNOWN_SOCKET_TYPE
* ERROR_REDIRECT_LOOP
* ERROR_OFFLINE
* ERROR_PORT_BLOCKED
* ERROR_NET_RESET
* ERROR_UNSAFE_CONTENT_TYPE
* ERROR_CORRUPTED_CONTENT
* ERROR_CONTENT_CRASHED
* ERROR_INVALID_CONTENT_ENCODING
* ERROR_UNKNOWN_HOST
* ERROR_MALFORMED_URI
* ERROR_UNKNOWN_PROTOCOL
* ERROR_FILE_NOT_FOUND
* ERROR_FILE_ACCESS_DENIED
* ERROR_PROXY_CONNECTION_REFUSED
* ERROR_UNKNOWN_PROXY_HOST

### Acceptance Criteria (how do I know when I’m done?)
- Correct error codes are shown
- User can exit error scenario and continue browsing to other websites

**Comment by bbinto** [comment]
As requested, I've created a telemetry bug for this #1242

**Comment by sblatz** [comment]
A-C does not have a way to add images to error messages right now which could get a little complicated due to us having to decode the image into a string and inject it into the HTML which can lead to huge performance issues. I've filed a ticket [here](https://github.com/mozilla-mobile/android-components/issues/2689).

**Comment by sblatz** [comment]
Even without the images, this will likely be closer to a 5. The android components architecture is just not set up in a way that makes it easy for us to do thinks like grab the app name or the URL which means we have to submit an a-c change regardless. I was able to style the pages fairly easily, but changing all the strings and adding the necessary "backend" work is a bit more effort.

**Comment by mheubusch** [comment]
I am fine to postpone images and new copy for post-MVP.  Are the styled pages visible in Nightly yet? That would help me determine if they are ok as is or figure out how important it is to match the mocks.

**Comment by sblatz** [comment]
@mheubusch They are not yet, I was waiting to merge before I had the "OK." This is an example of what one looks like without the image:

<img width=300 src=https://user-images.githubusercontent.com/4400286/55984707-e294ff80-5c52-11e9-9287-e4f095f85cb0.jpg>

**Comment by mheubusch** [comment]
@sblatz Ok to implement as is without matching design comps or adding illustrations. Please move forward and we can make the design updates post-MVP.

**Comment by kbrosnan** [comment]
Blocked on #1882 / https://github.com/mozilla-mobile/android-components/issues/2725 this issue is not ready for testing

**Comment by bifleming** [comment]
@sblatz is this blocked because there are bugs to fix?

**Comment by sblatz** [comment]
@bifleming https://github.com/mozilla-mobile/android-components/issues/2725 Needs to be fixed by a-c before QA can test this feature completely. The error pages show for everything **but** the most critical error pages (phishing attempts, etc.). This needs to be prioritized by a-c.

**Comment by sv-ohorvath** [comment]
Since safe browsing errors are moved to a new issue: #2397, this can be closed.
